packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -249,6 +261,8 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
     _animation = createAnimation()
       ..addStatusListener(_handleStatusChanged);
     assert(_animation != null, '$runtimeType.createAnimation() returned null.');
+    _performanceModeRequestHandle =
```

Comment:
This should probably move to _handleStatusChanged and execute when the animationstatus changes to fordward/reverse so you get the high performance mode again when the route is dismissed and that animation plays?